### PR TITLE
Utilize new archival options

### PR DESCRIPTION
**Pull request by mmitche**
Only archive when failing, rather than always.

/cc @jaredpar 

@jaredpar this will also need to be integrated to the other roslyn branches

### CODE REVIEW

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by jaredpar** [comment]
CC @jasonmalinowski

**Comment by jaredpar** [comment]
@mmitche looks like a real break

```
15:13:11 workspace://dotnet_roslyn%2Fmaster%2FGenPRTest%2Fgenerator_prtest/roslyn/netci.groovy: 45: unable to resolve class ArchivalSettings 
15:13:11  @ line 45, column 25.
15:13:11      def archiveSettings = new ArchivalSettings()
```

File: netci.groovy [hunk 1]
```diff
@@ -42,9 +42,21 @@ static String generateTriggerPhrase(String jobName, String opsysName, String tri
...
+  archiveSettings.addFiles('Binaries/**/*.pdb')
+  archiveSettings.addFiles('Binaries/**/*.xml')
+  archiveSettings.addFiles('Binaries/**/*.log')
+  archiveSettings.addFiles('Binaries/**/*.dmp')
+  archiveSettings.addFiles('Binaries/**/*.zip')
+  archiveSettings.addFiles('Binaries/**/*.png')
+  archiveSettings.addFiles('Binaries/**/*.xml')
+  archiveSettings.excludeFiles('Binaries/Obj/**')
+  archiveSettings.excludeFiles('Binaries/Bootstrap/**')
+  archiveSettings.excludeFiles('Binaries/**/nuget*.zip')
```
Comment (jasonmalinowski) [inline netci.groovy]: This no longer exists in this branch, as an FYI.

Comment (jaredpar) [inline netci.groovy]: At least part of my day will be happy seeing that line get deleted.

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by jaredpar** [comment]
What does sdf stand for?

**Comment by mmitche** [comment]
My random commit message when I'm just trying to get something to push/work

**Comment by jaredpar** [comment]
Why not asdf though?  At least then you're using your entire left hand.  Do you hate your pinky???

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by jaredpar** [comment]
@mmitche take a look at this generated job

http://dotnet-ci.cloudapp.net/job/dotnet_roslyn/job/master/job/GenPRTest/job/windows_debug_unit32/configure

Under the Flexible Publish section the following option is checked:

> Treat include and exclude patterns as case sensitive

Shouldn't that be unchecked?

**Comment by mmitche** [comment]
@jaredpar That's what it was set to before

**Pull request merged**